apps/ledger-live-mobile/src/contentCards/cards/vertical/elements.tsx

```diff
@@ -33,20 +35,55 @@ type ImageProps = {
   uri: string;
   size: Size;
   filledImage?: boolean;
+  isMediaOnly?: boolean;
+  mediaType: "image" | "video" | "gif";
 };
 
-export const Image = ({ uri, size, filledImage }: ImageProps) => {
+export const Media = ({ uri, size, filledImage, isMediaOnly, mediaType }: ImageProps) => {
   const isBigCardAndFilled = (size === "L" && filledImage) || false;
-  const stylesBigCard = isBigCardAndFilled
-    ? { marginBottom: 0, marginTop: 0, borderTopLeftRadius: 12, borderTopRightRadius: 12 }
+  const isMediumCardAndFilled = (size === "M" && filledImage) || false;
+  const isBigOrMediumCardAndFilled = isBigCardAndFilled || isMediumCardAndFilled;
+
+  const mediaOnlyStyle = isMediaOnly &&
+    isBigCardAndFilled && {
+      height: 282,
+      maxHeight: 282,
+      marginBottom: 0,
+      marginTop: 0,
+      borderRadius: 12,
+    };
+
+  const extraStylesCard = isBigOrMediumCardAndFilled
+    ? {
+        marginBottom: 0,
+        marginTop: 0,
+        borderTopLeftRadius: 12,
+        borderTopRightRadius: 12,
+        borderBottomLeftRadius: isMediumCardAndFilled ? 12 : 0,
+        borderBottomRightRadius: isMediumCardAndFilled ? 12 : 0,
+      }
     : { aspectRatio: 1 };
 
   return (
-    <NativeImage
-      source={{ uri }}
-      style={{ ...ImageStyles[size], ...stylesBigCard }}
-      resizeMode={isBigCardAndFilled ? "cover" : "contain"}
-    />
+    <>
```

Comment:
Do we need the fragment here ? We could do check the mediatype outside the return instruction and do two return instead (one for the video and one for the fastimage)

Reply:
Changed ✅